hazelcast/src/main/java/com/hazelcast/internal/journal/EventJournalReadOperation.java

```diff
@@ -86,6 +84,20 @@ public void beforeRun() {
 
         final int partitionId = getPartitionId();
         journal.cleanup(namespace, partitionId);
+
+        final long oldestSequence = journal.oldestSequence(namespace, partitionId);
+        final long newestSequence = journal.newestSequence(namespace, partitionId);
+
+        // fast forward if late
+        if (startSequence < oldestSequence) {
```

Comment:
can `startSequence` be in store? if yes, we are losing available data?

Reply:
Good catch, added additional check.